### PR TITLE
feat(suite-native): retrieving balances takes longer

### LINKED ISSUE #10151: Inform user when discovery takes too long
As a user I would like to be informed that the discovery takes more time than usual in order to know what is happening and won't be confused. 

Acceptance criteria:
1. When it takes ore than 45s to finish discovery of all assets, we display in Graph loading this message: "It takes longer than usual to retrieve balances. This may be caused by an unstable Internet connection."

### PR DESCRIPTION
**Pull request by juriczech**
## Related Issue

Resolve https://github.com/trezor/trezor-suite/issues/10151

## Screenshots:

https://github.com/trezor/trezor-suite/assets/36101761/de887c39-1a73-4b61-914b-403cc632efc2

### CODE REVIEW

File: suite-native/discovery/src/useIsDiscoveryDurationTooLong.tsx [hunk 1]
```diff
@@ -0,0 +1,32 @@
...
+    useEffect(() => {
+        if (isDiscoveryActive && startDiscoveryTimestamp) {
+            const interval = setInterval(() => {
+                if (performance.now() - startDiscoveryTimestamp > DISCOVERY_DURATION_TRESHOLD) {
+                    setLoadingTakesLongerThanExpected(true);
+                    clearInterval(interval);
+                }
+            }, DISCOVERY_LENGTH_CHECK_INTERVAL);
+
+            return () => clearInterval(interval);
```
Comment (Nodonisko) [inline suite-native/discovery/src/useIsDiscoveryDurationTooLong.tsx]: You shouldn't have cleanup function inside of `if`

**Review by Nodonisko — CHANGES_REQUESTED** [review]

File: suite-native/module-home/src/components/PortfolioGraph.tsx [hunk 2]
```diff
@@ -18,6 +19,9 @@ export const PortfolioGraph = () => {
     const fiatCurrency = useSelector(selectFiatCurrency);
     const isDiscoveryActive = useSelector(selectIsDeviceDiscoveryActive);
     const isPortfolioEmpty = useSelector(selectIsPortfolioEmpty);
+
+    const loadingTakesLongerThanExpected = useIsDiscoveryDurationTooLong();
```
Comment (Nodonisko) [inline suite-native/module-home/src/components/PortfolioGraph.tsx:23]: Can you please verify that this component doesn't rerender every second now?

File: suite-native/discovery/src/useIsDiscoveryDurationTooLong.tsx (hunk 1 above)
Comment (juriczech) [reply suite-native/discovery/src/useIsDiscoveryDurationTooLong.tsx]: Good point. Should be fixed now @Nodonisko

File: suite-native/module-home/src/components/PortfolioGraph.tsx (hunk 2 above)
Comment (juriczech) [reply suite-native/module-home/src/components/PortfolioGraph.tsx:23]: Verified. It doesn't rerender every second

**Review by Nodonisko — APPROVED** [review]

**Review by Nodonisko — APPROVED** [review]

**Pull request merged**